modules/apps/27-interchain-accounts/host/keeper/events.go

```diff
@@ -1,20 +1,29 @@
 package keeper
 
 import (
+	"fmt"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
 
 	icatypes "github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts/types"
 	"github.com/cosmos/ibc-go/v3/modules/core/exported"
 )
 
-// EmitWriteErrorAcknowledgementEvent emits an event signalling an error acknowledgement and including the error details
-func EmitWriteErrorAcknowledgementEvent(ctx sdk.Context, packet exported.PacketI, err error) {
+// EmitAcknowledgementEvent emits an event signalling a successful or failed acknowledgement and including the error
+// details if any.
+func EmitAcknowledgementEvent(ctx sdk.Context, packet exported.PacketI, ack exported.Acknowledgement, err error) {
+	errorMsg := ""
```

Comment:
nit: should we favour a var declaration here? 

```suggestion
	var errorMsg string
```